odoo/orm/models.py

```diff
@@ -4057,24 +4099,29 @@ def _determine_fields_to_fetch(self, field_names, ignore_when_in_cache=False) ->
         :param field_names: the list of fields requested
         :param ignore_when_in_cache: whether to ignore fields that are alreay in cache for ``self``
         :return: the list of fields that must be fetched
+        :raise AccessError: when trying to fetch fields to which the user does not have access
         """
         if not field_names:
             return []
 
         cache = self.env.cache
 
         fields_to_fetch = []
-        field_names_todo = deque(self.check_field_access_rights('read', field_names))
-        field_names_done = {'id'}  # trick: ignore 'id'
-
-        while field_names_todo:
-            field_name = field_names_todo.popleft()
-            if field_name in field_names_done:
+        field_todo = deque()
+        field_done = {self._fields['id']}  # trick: ignore 'id'
```

Comment:
Nitpicking on variable names...
```suggestion
        fields_todo = deque()
        fields_done = {self._fields['id']}  # trick: ignore 'id'
```